### PR TITLE
[Core][VariableUtils] DOFs and DOFs-reaction list addition at once

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
With this PR I'm extending the DOFs capabilities of the `VariableUtils` by adding the `AddDofsList` and `AddDofsListWithReactions` methods. The idea is to pass a list of strings with the variable names to be added as DOFs (e.g. `["DISPLACEMENT_X", "DISPLACEMENT_Y"]` or a list of DOF-reaction pairs (e.g. `[["DISPLACEMENT_X","REACTION_X"],["DISPLACEMENT_Y","REACTION_Y"]]`).

This makes possible to add all the DOFs at once, or in other words, avoids looping all the nodes for each DOFs as we're doing so far.

Nice thing is that, in the long term, the DOFs addition could be automated by retrieving them from the elements and conditions (see #9484).

### CODE REVIEW

File: kratos/python/add_variable_utils_to_python.cpp [hunk 1]
```diff
@@ -462,6 +462,8 @@ void AddVariableUtilsToPython(pybind11::module &m)
         .def("SumElementVectorVariable", &VariableUtils::SumElementVectorVariable)
         .def("AddDof", &VariableUtils::AddDof<Variable<double>>)
         .def("AddDof", &VariableUtils::AddDofWithReaction<Variable<double>>)
+        .def_static("AddDofsList", &VariableUtils::AddDofsList)
+        .def_static("AddDofsList", &VariableUtils::AddDofsWithReactionsList)
```
Comment (philbucher) [inline kratos/python/add_variable_utils_to_python.cpp:466]:
```suggestion
        .def_static("AddDofsWithReactionsList", &VariableUtils::AddDofsWithReactionsList)
```

File: kratos/tests/test_variable_utils.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ﻿# Importing the Kratos Library
+from unittest.case import expectedFailure
```
Comment (philbucher) [inline kratos/tests/test_variable_utils.py]: where is this used?

File: kratos/utilities/variable_utils.cpp [hunk 3]
```diff
@@ -115,6 +115,67 @@ array_1d<double, 3> VariableUtils::SumElementVectorVariable(
 /***********************************************************************************/
 /***********************************************************************************/
 
+void VariableUtils::AddDofsList(
+    const std::vector<std::string>& rDofsVarNamesList,
+    ModelPart& rModelPart)
+{
+    // Create a set with the variables to be added as DOFs
+    std::set<const Variable<double>*> dofs_vars_set;
```
Comment (philbucher) [inline kratos/utilities/variable_utils.cpp]:
```suggestion
    std::unordered_set<const Variable<double>*> dofs_vars_set;
```

File: kratos/utilities/variable_utils.h [hunk 4]
```diff
@@ -1336,6 +1336,14 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
         KRATOS_CATCH("")
     }
 
+    static void AddDofsList(
```
Comment (philbucher) [inline kratos/utilities/variable_utils.h:1346]: doc is missing

**Review by philbucher — COMMENTED** [review]
minor otherwise :+1:

File: kratos/python/add_variable_utils_to_python.cpp (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/python/add_variable_utils_to_python.cpp:466]: I followed the same criterion that our current `AddDofs` has.

File: kratos/tests/test_variable_utils.py (hunk 2 above)
Comment (rubenzorrilla) [reply kratos/tests/test_variable_utils.py]: No idea. Actually I don't know why it appears as a change.

File: kratos/utilities/variable_utils.h (hunk 4 above)
Comment (rubenzorrilla) [reply kratos/utilities/variable_utils.h:1346]: Sure. Thanks for spotting.

**Review by philbucher — DISMISSED** [review]
👍

File: kratos/tests/test_variable_utils.py (hunk 2 above)
Comment (rubenzorrilla) [reply kratos/tests/test_variable_utils.py]: Removed. Most probably I added it by accident.

File: kratos/utilities/variable_utils.h (hunk 4 above)
Comment (rubenzorrilla) [reply kratos/utilities/variable_utils.h:1346]: Done.

**Comment by rubenzorrilla** [comment]
Note that to also use an `unordered_set` in the DOFs with reaction function I required to add the hasher and compare function (not sure at all if it is really worth it in this case but already done :sweat_smile:). Also note that I made the provided variables check more exhaustive.

**Review by philbucher — APPROVED** [review]

**Pull request merged**